client/apps/edit/components/content/article_layouts/video.jsx

```diff
@@ -0,0 +1,225 @@
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { VideoAbout, VideoAboutContainer } from '@artsy/reaction-force/dist/Components/Publishing/Video/VideoAbout'
+import { VideoCover, VideoCoverContainer } from '@artsy/reaction-force/dist/Components/Publishing/Video/VideoCover'
+import FileInput from '/client/components/file_input/index.jsx'
+import Paragraph from '/client/components/rich_text/components/paragraph.coffee'
+import { PlainText } from '/client/components/rich_text/components/plain_text'
+import { ProgressBar } from '/client/components/file_input/progress_bar.jsx'
+import { IconRemove } from '@artsy/reaction-force/dist/Components/Publishing/Icon/IconRemove'
+
+export class EditVideo extends Component {
+  static propTypes = {
+    article: PropTypes.object.isRequired,
+    onChange: PropTypes.func.isRequired
+  }
+
+  state = {
+    uploadProgress: null,
+    showVideo: false
+  }
+
+  onMediaChange = (key, value) => {
+    const { article, onChange } = this.props
+    const media = article.get('media') || {}
+
+    media[key] = value
+    onChange('media', media)
+  }
+
+  editDescription = () => {
+    const { article, onChange } = this.props
+
+    return (
+      <PlainText
+        content={article.get('description')}
+        onChange={(key, value) => onChange('description', value)}
+        placeholder='Description'
+        name='description'
+      />
+    )
+  }
+
+  editTitle = () => {
+    const { article, onChange } = this.props
+
+    return (
+      <PlainText
+        content={article.get('title')}
+        onChange={(key, value) => onChange('title', value)}
+        placeholder='Title'
+        name='title'
+      />
+    )
+  }
+
+  editMediaDescription = () => {
+    const { article } = this.props
+    const description = article.get('media') && article.get('media').description
+
+    return (
+      <Paragraph
+        html={description || ''}
+        linked
+        onChange={(html) => this.onMediaChange('description', html)}
+        placeholder='Start writing here...'
+      />
+    )
+  }
+
+  editMediaCredits = () => {
+    const { article } = this.props
+    const credits = article.get('media') && article.get('media').credits
+
+    return (
+      <Paragraph
+        html={credits || ''}
+        linked
+        onChange={(html) => this.onMediaChange('credits', html)}
+        placeholder='Start writing here...'
+      />
+    )
+  }
+
+  render () {
+    const { article } = this.props
+    const { uploadProgress } = this.state
+    const media = article.get('media') || {}
+
+    return (
+      <EditVideoContainer>
+        {uploadProgress &&
+          <ProgressBar progress={uploadProgress} />
+        }
+
+        <VideoPreview visible={this.state.showVideo}>
+          <video
+            controls
+            src={media.url}
+            onLoadedMetadata={(e) => this.onMediaChange('duration', e.target.duration)}
+          />
```

Comment:
Also note that I've switched to using the plain `video` element instead of the player because we need a callback for `onLoadedMetadata` which I realized is not present in the Reaction version of VideoPlayer. Will update that soon after Tuesday.